mlflow/store/file_store.py

```diff
@@ -585,6 +585,21 @@ def set_tag(self, run_id, tag):
         # Don't add trailing newline
         write_to(tag_path, self._writeable_value(tag.value))
 
+    def delete_tag(self, run_id, key):
+        """
+        Delete a tag from a run.
+        :param run_id: ID of the run that contains the tag to delete
+        :param key: name of the tag to delete
+        """
+        _validate_run_id(run_id)
+        run = self.get_run(run_id)
+        check_run_is_active(run.info)
+        if key not in run.data.tags.keys():
+            raise MlflowException("No tag with name: {} in run with id {}".format(key, run_id),
+                                  error_code=RESOURCE_DOES_NOT_EXIST)
+        tag_path = self._get_tag_path(run.info.experiment_id, run_id, key)
+        mv(tag_path, self.trash_folder)
```

Comment:
The trash folder is currently used to store deleted experiments, which may be restored by the user at a later date. I am concerned about adding deleted tags to the same folder because we risk misinterpreting them as experiments at some point (currently, every subdirectory of `.trash` is assumed to be an experiment; we don't have any assumptions about files because files don't currently exist at the top level of `.trash`).

Additionally, what happens if the user deletes two tags with the same key and different values from different runs? `.trash` will only keep the most recently deleted tag because the key is used as the file name.

Based on this concerns, I think it may be simpler to just delete the file, rather than moving it to `.trash`.

Reply:
Thanks for pointing that out! Editing this to just use shutil.rmtree which should just delete instead.